### PR TITLE
143 style number of results in search screen

### LINKED ISSUE #143: Style number of results in SearchScreen

### PR DESCRIPTION
**Pull request by haakonjacobsen**
Resolve #143

### CODE REVIEW

File: frontend/screens/SearchScreen.tsx [hunk 1]
```diff
@@ -85,75 +87,81 @@ function SearchScreen() {
...
+            style={styles.bottomGradient}
           />
+        </View>
+        {filter.dogBreeds.length > 0 && !!data ? (
+          displayType !== 'Map' ? (
+            <FlashList
+              ref={searchRef}
+              numColumns={displayType === 'Grid' ? 2 : 1}
+              refreshing={isFetching}
+              onRefresh={() => refetchPosts()}
```
Comment (AndreasWintherMoen) [inline frontend/screens/SearchScreen.tsx:106]: Unnecessary anonymous function

```suggestion
              onRefresh={refetchPosts}
```

File: frontend/screens/SearchScreen.tsx [hunk 2]
```diff
@@ -85,75 +87,81 @@ function SearchScreen() {
...
+              data={data?.pages.flat()}
+              keyExtractor={(item) => item._id}
+              renderItem={({ item }) => (
+                <GenericCard cardType={displayTypeDict[displayType]} key={item._id} post={item} />
+              )}
+              onEndReached={() => fetchNextPage()}
```
Comment (AndreasWintherMoen) [inline frontend/screens/SearchScreen.tsx:112]: Same here

```suggestion
              onEndReached={fetchNextPage}
```

File: frontend/screens/SearchScreen.tsx [hunk 3]
```diff
@@ -85,75 +87,81 @@ function SearchScreen() {
...
           />
+        </View>
+        {filter.dogBreeds.length > 0 && !!data ? (
+          displayType !== 'Map' ? (
+            <FlashList
+              ref={searchRef}
+              numColumns={displayType === 'Grid' ? 2 : 1}
+              refreshing={isFetching}
+              onRefresh={() => refetchPosts()}
+              data={data?.pages.flat()}
```
Comment (AndreasWintherMoen) [inline frontend/screens/SearchScreen.tsx:107]: Consider putting this in a useMemo if data can be large.

Comment (AndreasWintherMoen) [reply frontend/screens/SearchScreen.tsx:107]: I see it's used multiple places, so it may not be a bad idea to make it a variable anyway.

**Review by AndreasWintherMoen — COMMENTED** [review]
Some minor stuff you should look at.

File: frontend/screens/SearchScreen.tsx (hunk 3 above)
Comment (haakonjacobsen) [reply frontend/screens/SearchScreen.tsx:107]: Do you mean the refetchPosts() should be put in a useMemo?

Comment (haakonjacobsen) [reply frontend/screens/SearchScreen.tsx]: Done

**Comment by haakonjacobsen** [comment]
Fixed! No time for new review!😎
bitmoji

**Pull request merged**